packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -87,45 +99,41 @@ function handleRequireCallsToNodeJSBuiltins(build: PluginBuild) {
 	);
 }
 
+/**
+ * Handles aliased NPM packages.
+ *
+ * @param build ESBuild PluginBuild.
+ * @param unresolvedAlias Unresolved aliases from the presets.
+ * @param alias Aliases resolved to absolute paths.
+ * @param external external modules.
+ */
 function handleUnenvAliasedPackages(
 	build: PluginBuild,
+	unresolvedAlias: Record<string, string>,
 	alias: Record<string, string>,
 	external: string[]
 ) {
-	// esbuild expects alias paths to be absolute
-	const aliasAbsolute: Record<string, string> = {};
-	for (const [module, unresolvedAlias] of Object.entries(alias)) {
-		try {
-			aliasAbsolute[module] = require
-				.resolve(unresolvedAlias)
-				.replace(/\.cjs$/, ".mjs");
-		} catch (e) {
-			// this is an alias for package that is not installed in the current app => ignore
-		}
-	}
-
-	const UNENV_ALIAS_RE = new RegExp(
-		`^(${Object.keys(aliasAbsolute).join("|")})$`
-	);
+	const UNENV_ALIAS_RE = new RegExp(`^(${Object.keys(alias).join("|")})$`);
 
 	build.onResolve({ filter: UNENV_ALIAS_RE }, (args) => {
-		const unresolvedAlias = alias[args.path];
+		const unresolved = unresolvedAlias[args.path];
 		// Convert `require()` calls for NPM packages to a virtual ES Module that can be imported avoiding the require calls.
 		// Note: Does not apply to Node.js packages that are handled in `handleRequireCallsToNodeJSBuiltins`
 		if (
 			args.kind === "require-call" &&
-			(unresolvedAlias.startsWith("unenv/runtime/npm/") ||
-				unresolvedAlias.startsWith("unenv/runtime/mock/"))
+			(unresolved.startsWith("unenv/runtime/npm/") ||
+				unresolved.startsWith("unenv/runtime/mock/"))
 		) {
 			return {
 				path: args.path,
 				namespace: REQUIRED_UNENV_ALIAS_NAMESPACE,
 			};
 		}
+
 		// Resolve the alias to its absolute path and potentially mark it as external
 		return {
-			path: aliasAbsolute[args.path],
-			external: external.includes(unresolvedAlias),
+			path: alias[args.path],
+			external: external.includes(unresolved),
```

Comment:
Shoudn't the `external` paths already be resolved?
In which case why use `unresolved` here, rather than `path`?

Reply:
TBH I don't fully get what the code is doing here...

Reply:
I tried to update the code but revert to the original behavior